src/components/chip/chip.tsx

```diff
@@ -0,0 +1,279 @@
+
+    /**
+     * ID of the chip. Must be unique.
+     */
+    @Prop()
+    id: number | string;
+
+    /**
+     * Label displayed on the chip
+     */
+    @Prop()
+    public text: string;
+
+    /**
+     * Icon of the chip.
+     */
+    @Prop()
+    public icon: string | Icon;
+
+    /**
+     * If supplied, the chip will become a clickable links.
+     */
+    @Prop()
+    link?: Omit<Link, 'text'>;
+
+    /**
+     * The value of the badge, displayed on the chip.
+     */
+    @Prop()
+    badge?: string | number;
+
+    /**
+     * Set to `true` to disable the chip.
+     */
+    @Prop({ reflect: true })
+    public disabled = false;
+
+    /**
+     * Set to `true` to render the chip as a static UI element.
+     * Useful when the parent component has a `readonly` state.
+     */
+    @Prop({ reflect: true })
+    public readonly = false;
+
+    /**
+     * Set to `true` to visualize the chip in a "selected" state.
+     * This is typically used when the chip is used in a chip-set
+     * along with other chips.
+     */
+    @Prop({ reflect: true })
+    public selected = false;
+
+    /**
+     * Set to `true` to render a remove button on the chip.
+     */
+    @Prop({ reflect: true })
+    public removable = false;
+
+    /**
+     * Set to `filter` to render the chip with a distinct style
+     * suitable for visualizing filters.
+     */
+    @Prop({ reflect: true })
+    public type: 'filter';
+
+    /**
+     * Fired when clicking on buttons (not links!)
+     * inside the breadcrumbs.
+     */
+    @Event()
+    public remove: EventEmitter<Chip>;
+
+    @Element()
+    private host: HTMLLimelChipElement;
+
+    private button: HTMLButtonElement;
+    private anchor: HTMLAnchorElement;
+
+    public componentWillLoad() {
+        makeEnterClickable(this.host as HTMLElement);
+    }
+
+    public disconnectedCallback() {
+        this.removeEnterClickable();
+    }
+
+    public render() {
+        if (this.link) {
+            return this.renderAsLink();
+        }
+
+        return this.renderAsButton();
+    }
+
+    private renderAsButton = () => {
+        return [
+            <button
+                class="chip"
+                role="button"
+                disabled={this.disabled || this.readonly}
+                onKeyDown={this.handleDeleteKeyDown}
+            >
+                {this.renderIcon()}
+                {this.renderLabel()}
+                {this.renderBadge()}
+            </button>,
+            this.renderRemoveButton(),
+        ];
+    };
+
+    private renderAsLink = () => {
+        return [
+            <a
+                class="chip"
+                href={this.link.href}
+                title={this.link.title}
+                aria-disabled={this.disabled || this.readonly}
+                tabindex={this.disabled ? -1 : 0}
+                onClick={(e) => {
+                    if (this.disabled) {
+                        e.preventDefault();
+                    }
+                }}
```

Comment:
Using arrow-functions in templates (or, rather, _defining_ arrow-functions in templates) is bad for performance, since it creates a new function every time the component is re-rendered. So this should be extracted to something like:

```tsx
                onClick={this.filterClick}
    …

    private filterClick = (e: Event) => {
        if (this.disabled) {
            e.preventDefault();
        }
    };
```